CONTRIBUTE.md

```diff
@@ -0,0 +1,177 @@
+1.  Fork the Ganeti repo on GitHub
+2.  Make changes in a feature branch, probably based off the master branch.
+2.  Test using "make pylint; make hlint; make py-tests; make hs-tests"
+3.  Send your changes for review by opening a Pull Request.
+4.  Sign the [Google CLA](https://cla.developers.google.com/).
+5.  Address the reviewer comments by making new commits and ask PTAL.
+6.  Wait for your Pull Request to be accepted.
+
+### 1. Forking the Repo
+
+The Ganeti project uses Pull Requests to accept contributions. In order to do so
+you must first [fork](https://help.github.com/articles/fork-a-repo/) the Ganeti
+repo: https://github.com/ganeti/ganeti. After that, clone your fork and add the
+original repository as the upstream:
+
+```
+git clone git@github.com:<username>/ganeti.git
+git remote add upstream git@github.com:ganeti/ganeti.git
+```
+
+### 2. Merging Upstream changes
+
+Important: Before starting a new patch it is a good idea to
+[merge all the upstream changes](https://help.github.com/articles/syncing-a-fork/):
+
+```
+git fetch upstream
+git checkout <branch>
+git merge upstream/<branch>
+```
+
+### 3. Creating your feature branch
+We recommend using
+[feature branches](https://www.atlassian.com/git/tutorials/using-branches)
+in your repo instead of commit directly to the master and/or stable branches.
+Without feature branches you could have a local clone in a separate folder named
+after the feature. This would work well as long as you don't want to make
+parallel pull requests or backup your local work on github. If your local
+folders map to respective feature branches on your personal fork on github, you
+can work in parallel on multiple features with backing up local commits on
+github as well as work on multiple pull requests.
+
+Let's say you want to patch an existing issue on branch <existing_branchname>.
+Then:
+```
+git checkout <existing_branchname>
+git checkout -b <feature_branchname>
+```
+
+Try to use a descriptive name for the feature branch. Names like
+"feature-<issue_id>", "bug-<issue_id>" are good when referring to an existing
+bug/feature request. Using a name related to the theme of the changes is also a
+good idea (i.e.: "predictive_locking").
+
+### 4. Making your changes
+
+Make the changes you need/want to do on your feature branch. Make as many
+commits as you want, making sure that the commit messages are small and concise
+within the context of your changes. The commit messages are going to be useful
+for the reviewer, so be reasonable. Favor small Pull Requests as they are easier
+to review and they provide feedback early. Avoid making huge changes just to see
+the reviewer asking you to restart from scratch..
+
+Remember to use the --signoff flag when committing. You may also push these
+changes to your personal fork by running `git push origin <feature_branchname>`
+
+### 5. Testing your changes
+
+`make pylint; make hlint; make py-tests; make hs-tests`
```

Comment:
Here (and above), it's okay at the moment but we do have more options for testing and sanity checking. The problem is that they've been pretty bitrotten lately and someone should take a look at them again (see: make distcheck). For the moment, these four options are all we've been doing so this is okay.